google-cloud-logging/src/test/java/com/google/cloud/logging/LoggingImplTest.java

```diff
@@ -2304,6 +2311,22 @@ public void testDiagnosticInfoWithPartialSuccess() {
     testDiagnosticInfoGeneration(true);
   }
 
+  @Test
+  public void testPartialSuccessNotOverridenIfPresent() {
+    WriteLogEntriesRequest request =
+        WriteLogEntriesRequest.newBuilder()
+            .addAllEntries(
+                Iterables.transform(
+                    ImmutableList.of(LOG_ENTRY1, LOG_ENTRY2), LogEntry.toPbFunction(PROJECT)))
+            .setPartialSuccess(false)
+            .build();
+    WriteLogEntriesResponse response = WriteLogEntriesResponse.getDefaultInstance();
+    EasyMock.expect(loggingRpcMock.write(request)).andReturn(ApiFutures.immediateFuture(response));
+    EasyMock.replay(rpcFactoryMock, loggingRpcMock);
+    logging = options.getService();
+    logging.write(ImmutableList.of(LOG_ENTRY1, LOG_ENTRY2), WriteOption.partialSuccess(false));
+  }
```

Comment:
should there be a duplicate test with `setPartialSuccess(true)`?

Reply:
There are bunch of other tests in this file doing that

Reply:
Are you sure? The only other test I see that calls `logging.write` with a partial success option explicitly set is `testDiagnosticInfoGeneration`, but that's testing the case with a diagnostic entry, which follows a different flow through the code. 

It may be overkill, but it would be good to cover all the cases with the tests to be safe. Since the logic is a bit tricky here

Reply:
New functionality is tested when `partialSuccess` it is not overriden if present in `options` - I believe this case is exactly tested here and it is enough since we do not care about a value, but rather that it is present or not.